status-core/src/main/java/com/indeed/status/core/AbstractDependencyManager.java

```diff
@@ -284,7 +289,11 @@ public void launchPinger(final Dependency dependency) {
         // unpredictably.
         pinger.addListener(updateHandler);
 
-        executor.scheduleWithFixedDelay(pinger, 0, pinger.getPingPeriod(), TimeUnit.MILLISECONDS);
+        // When no pinger already exists, schedule the pinger and add it to the collection so it can be removed if/when
+        // AbstractDependencyManager#removeDependency is called. We don't need to throw an error
+        // for duplicate dependencies since that is handled in AbstractDependencyManager#addDependency
```

Comment:
This comment is excessive as it simply explains the line of code below. I think you can leave a comment along the lines of "no need to check for duplicates since that is handled in #addDependency"